mllib/src/test/scala/org/apache/spark/ml/tree/impl/RandomForestSuite.scala

```diff
@@ -640,4 +740,55 @@ private object RandomForestSuite {
     val (indices, values) = map.toSeq.sortBy(_._1).unzip
     Vectors.sparse(size, indices.toArray, values.toArray)
   }
+
+  /** Generate a label. */
+  private def generateLabel(rnd: Random, numClasses: Int): Double = {
+    rnd.nextInt(numClasses)
+  }
+
+  /** Generate a numeric value in the range [numericMin, numericMax]. */
+  private def generateNumericValue(rnd: Random, numericMin: Double, numericMax: Double) : Double = {
+    rnd.nextDouble() * (numericMax- numericMin) + numericMin
+  }
+
+  /** Generate a binary value. */
+  private def generateBinaryValue(rnd: Random) : Double = if (rnd.nextBoolean()) 1 else 0
+
+  /** Generate an array of binary values of length numBinary. */
+  private def generateBinaryArray(rnd: Random, numBinary: Int): Array[Double] = {
+    Range.apply(0, numBinary).map(_ => generateBinaryValue(rnd)).toArray
```

Comment:
`Array.fill(numBinary)(generateBinaryValue(rnd))` is simpler

Reply:
I have updated accordingly the generation of both binary and numeric values, it is indeed simpler this way